modules/onboarding/src/main/kotlin/edu/stanford/spezi/module/onboarding/consent/FirebasePdfService.kt

```diff
@@ -0,0 +1,148 @@
+package edu.stanford.spezi.module.onboarding.consent
+
+import android.graphics.Paint
+import android.graphics.Typeface
+import android.graphics.pdf.PdfDocument
+import android.text.Layout
+import android.text.StaticLayout
+import android.text.TextPaint
+import androidx.compose.ui.graphics.asAndroidPath
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.storage.FirebaseStorage
+import edu.stanford.spezi.core.design.component.markdown.MarkdownElement
+import edu.stanford.spezi.core.design.component.markdown.parseMarkdown
+import edu.stanford.spezi.core.logging.speziLogger
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.time.LocalDate
+import javax.inject.Inject
+
+class FirebasePdfService @Inject internal constructor(
```

Comment:
As noted in some of the other models we should aim to differentiate a Firebase implementation and connection with a generic set of components that allow integrations like this.
In the Swift module, we have actually engrained the PDF creation setup right in the module itself and don't provide a way to customize this from the outside, it is a core functionality but we offer a nice mechanism to get the finished PDF and then decide what you want do do with it (including the Firebase upload which is then implemented right in the app as it's a single function and hard to provide a generic way to implement this.

Reply:
yes that's a good point, i'll move the PDF export to a separate service, because otherwise it also violates the singe responseability concept of the class 👍

Reply:
Fully agree with this comment 👍 Have provided a similar comment in `OnboardingModule.kt`